test/fit_tests/tests/fileserver/test_microkernel_static_file_server.py

```diff
@@ -0,0 +1,368 @@
+    mon_url = '/microkernel?name='+filename
+    response =filerestful( "http://" + serverip + ":7070" + mon_url,rest_action="binary-put", rest_payload=file)
+    if response['status'] in range(200,205):
+         return response['json']
+    else:
+         print 'Incorrect HTTP return code, expected 201, got:' + str(response['status'])
+         return "fail"
+
+def list_microkernel():
+    mon_url = '/microkernel'
+    serverip=os.getenv("IMAGESERVER", "localhost")
+    response =filerestful( "http://" +serverip + ":7070" + mon_url)
+    if response['status'] in range(200,205):
+         return response['json']
+    else:
+         print 'Incorrect HTTP return code, expected 201, got:' + str(response['status'])
+         return "fail"
+
+def delete_microkernel(filename):
+    mon_url = '/microkernel?name='+filename
+    serverip=os.getenv("IMAGESERVER", "localhost")
+    response =filerestful( "http://" +serverip + ":7070" + mon_url,rest_action="delete")
+    if response['status'] in range(200,205):
+         return response['json']
+    else:
+         print 'Incorrect HTTP return code, expected 201, got:' + str(response['status'])
+         return "fail"
+
+# Check if the node is rediscovered. Retry in every 30 seconds and total 600 seconds.
+def wait_for_discover(node_uuid):
+    for dummy in range(0, 20):
+        fit_common.time.sleep(30)
+        rest_data=fit_common.rackhdapi('/redfish/v1/Systems/')
+        if rest_data['json']['Members@odata.count']==0:
+            continue
+        node_collection= rest_data['json']['Members']
+        for computenode in node_collection:
+            nodeidurl=computenode['@odata.id']
+            api_data=fit_common.rackhdapi(nodeidurl)
+            if api_data['status']>399:
+                break
+            if node_uuid== api_data['json']['UUID']:
+                return True
+    print "Timeout!"
+    return False
+
+def apply_obmsetting(nodeid):
+    usr=''
+    pwd=''
+    response= fit_common.rackhdapi('/api/2.0/nodes/'+nodeid+'/catalogs/bmc')
+    bmcip= response['json']['data']['IP Address']
+    #Try credential record in config file
+    for creds in fit_common.GLOBAL_CONFIG['credentials']['bmc']:
+        if fit_common.remote_shell('ipmitool -I lanplus -H ' + bmcip+' -U ' + creds['username']+' -P '+ creds['password'] + ' fru')['exitcode'] == 0:
+            usr = creds['username']
+            pwd = creds['password']
+            break
+    # Put the credential to OBM settings
+    if  usr!="":
+        payload = { "service": "ipmi-obm-service","config": {"host": bmcip, "user": usr,"password": pwd},"nodeId": nodeid}
+        api_data = fit_common.rackhdapi("/api/2.0/obms", action='put', payload=payload)
+        if api_data['status']== 201:
+            return True
+    return False
+
+
+# Select test group here using @attr
+from nose.plugins.attrib import attr
+@attr(all=True, regression=True, smoke=True)
+class static_microkernel_file_server(fit_common.unittest.TestCase):

```

Comment:
How long does this test take to run, does it meet smoke criteria?
And can it be run against virtual/vargrant environment?
if not, change smoke=False.
thanks!